client/landing/stepper/declarative-flow/internals/steps-repository/flex-site-creation/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { Step } from '@automattic/onboarding';
+import { TextControl, Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import { FormEvent, useState } from 'react';
+import DocumentHead from 'calypso/components/data/document-head';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import type { Step as StepType } from '../../types';
+import './style.scss';
+
+const FlexSiteCreation: StepType< {
+	submits: {
+		siteName: string;
+	};
+} > = function FlexSiteCreation( { navigation } ) {
+	const { submit } = navigation;
+	const { __ } = useI18n();
+	const { setSiteTitle } = useDispatch( ONBOARD_STORE );
+
+	const [ siteName, setSiteName ] = useState( '' );
+	const [ isLoading, setIsLoading ] = useState( false );
+
+	const handleSubmit = ( event: FormEvent ) => {
+		event.preventDefault();
+
+		if ( ! siteName.trim() ) {
+			return;
+		}
+
+		setIsLoading( true );
+
+		// Store site title in ONBOARD_STORE so create-site step can use it
+		setSiteTitle( siteName );
```

Comment:
It would be great if you could set this value in the flow instead of here. This way the step remains more versatile and reusable. Now it has a side effect.